torch/csrc/jit/passes/symbolic_shape_analysis.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <torch/csrc/WindowsTorchApiMacro.h>
+#include <torch/csrc/jit/ir/ir.h>
+
+namespace torch {
+namespace jit {
+
+// CAUTION NOT TO BE USED, STILL A WIP, NOT STABLE
+
+TORCH_API void PropagateShapesWithShapeFunction(
+    Node* n,
+    const std::shared_ptr<Graph>& graph);
+
+TORCH_API void RegisterOperatorShapeFunction(
+    Node* n,
+    const std::shared_ptr<Graph>& graph);
```

Comment:
Nit: we might be more consistent about how we pass shared pointers. In this PR in different places they are passed by value, by reference, and by constant reference.